### PR TITLE
Fix race between replica entering ST and reserved page APIs

### PR DESCRIPTION
**Pull request by mdarade**
Fix race between replica entering ST and reserved page APIs

### CODE REVIEW

**Review by StefanPalashev — APPROVED** [review]
LGTM, thanks @mdarade.

File: bftengine/src/bcstatetransfer/BCStateTran.cpp [hunk 1]
```diff
@@ -663,9 +665,12 @@ void BCStateTran::saveReservedPage(uint32_t reservedPageId, uint32_t copyLength,
 
 // TODO(TK) check if this function can have its own transaction(bftimpl)
 void BCStateTran::zeroReservedPage(uint32_t reservedPageId) {
-  LOG_DEBUG(logger_, reservedPageId);
+  if (psd_->getIsFetchingState()) {
```
Comment (glevkovich) [inline bftengine/src/bcstatetransfer/BCStateTran.cpp:668]: `if (isCollectingState_) {`
is much faster.
I am going to make some changes (will upload PR) to make isCollectingState_ safer, although the idea of polling on the state is a very bad idea...and can always lead to wrong results.
===Same for line 648===

**Review by glevkovich — APPROVED** [review]
@mdarade - 2 small changes, use isCollectingState_ and merge.

**Comment by nityavasudevan** [comment]
Please add the details in Jira on the fix and testing that we did as part if the PR.

**Pull request merged**